packages/types/src/query-engine-base/index.ts

```diff
@@ -0,0 +1,94 @@
+import type { TypedDocumentNode } from '@graphql-typed-document-node/core';
+import type { DocumentNode } from 'graphql/language';
+import type { ComponentType, PropsWithChildren } from 'react';
+
+/**
+ * A generic interface for creating adapters to support different GraphQL clients
+ * both on the server and client side.
+ *
+ * @template TClient - The type of the GraphQL client instance.
+ * @template TClientOptions - The type of options used to initialize the client.
+ */
+export interface BaseQueryClientEngine<
```

Comment:
`Base` is redundant.

```suggestion
export interface QueryClientEngine<
```